pkg/workflows/sdk/v2/pb/sdk.proto

```diff
@@ -35,10 +35,10 @@ message ConsensusDescriptor {
 }
 
 message CapabilityRequest {
-  string executionId = 1;
```

Comment:
Why are you dropping the execution ID? Engine needs it to distinguish executions because it has only 1 host object that it receives capability calls from.

Reply:
It was blindly passed from the WASM. I'll look at doing it differently.

Reply:
There's a TODO to use it in the Engine, I'm not sure where the host plugs it in before your change...
I agree that it has to be done in the host, not guest.

Requiring a separate callback object per execution also sounds OK to me but it's a larger change. Why not keep things as they were?

Reply:
It's a lot cleaner for the engine to be in control of the ID. It's the one that's aware of why it's used (to correlate requests across instances).

Reply:
I think there's value in having the execution ID in the host as well. Even if only for debugging purposes (we usually grep logs for the execution ID and get nice history of everything that happened). If you can, please keep it there.

Reply:
We didn't use it for that yet. The right approach then is to return the ID in any errors in core. We can also have the logger be execution specific and decorate it. I'm a huge fan of separation of concerns. When we run workflows in TEEs, the less they need to understand the easier it'll be to isolate. We'll also keep user debugging cleaner in the simulator if it doesn't have bloat.

Reply:
SGTM. I don't mind modifying module.Execute() to take a callback handler, thanks.